src/naming/registration.cairo

```diff
@@ -52,14 +54,22 @@ func booked_domain(hashed_domain: felt) -> (booking_data: (owner: felt, expiry:
 }
 
 func pay_buy_domain{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    current_timestamp, days, caller, domain
+    current_timestamp, days, caller, domain, sponsor
 ) -> () {
+    let useless = 1;
     let (pricing_contract) = _pricing_contract.read();
     let (erc20, price) = Pricing.compute_buy_price(pricing_contract, domain, days);
     let (naming_contract) = get_contract_address();
     with_attr error_message("ERC20 transfer impossible: check your ETH balance") {
         IERC20.transferFrom(erc20, caller, naming_contract, price);
     }
+
+    if (sponsor != 0) {
+        let (referral_contract) = _referral_contract.read();
+        Referral.add_commission(referral_contract, price, sponsor);
```

Comment:
So the referral is about the address but to make it usable you need a domain right ?